src/main/java/weizberg/gameoflife/GridComponent.java

```diff
@@ -0,0 +1,112 @@
+package weizberg.gameoflife;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.util.Arrays;
+
+public class GridComponent extends JComponent {
+    private final Grid grid;
+    private int[][] field;
+    Timer timer = new Timer(1000, new ActionListener() {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            grid.nextGen();
+        }
+    });
+
+    public GridComponent(Grid grid) {
+        this.grid = grid;
+        addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                int gridX = e.getX() / 20;
+                int gridY = e.getY() / 20;
+
+                if (grid.isAlive(gridX, gridY)) {
+                    grid.makeDead(gridX, gridY);
+                } else {
+                    grid.makeAlive(gridX, gridY);
+                }
+                for (int y = 0; y < getWidth(); y++) {
+                    for (int x = 0; x < getHeight(); x++) {
+                        if (grid.isAlive(x, y)) {
+                            System.out.println(x + ", " + y + " is alive");
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void mousePressed(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseReleased(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseEntered(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseExited(MouseEvent e) {
+
+            }
+        });
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+
+        field = grid.getField();
+
+        g.setColor(Color.white);
+
+        for (int i = 0; i <= getWidth(); i += 20) {
+            g.drawLine(i, 0, i, getHeight());
+        }
+
+        for (int i = 0; i <= getHeight(); i += 20) {
+            g.drawLine(0, i, getWidth(), i);
+        }
+
+        g.setColor(Color.MAGENTA);
+
+        int rows = field.length;
+        int cols = field[0].length;
+        for (int y = 0; y < rows; y++) {
+            for (int x = 0; x < cols; x++) {
+                if (grid.isAlive(x, y)) {
+                    g.fillRect(x * 20, y * 20, 20, 20);
+                }
+            }
+        }
+        repaint();
```

Comment:
If you're using a timer then you don't need repaint here, put it in the timer.

Reply:
When I moved repaint() to the timer, it does not paint when I click on a cell.